app/components/SearchableMetricSelector.js

```diff
@@ -1,25 +1,75 @@
 import { useEffect, useState } from 'react'
 
-function Modal(props){
-  const { supportedMetrics, onClose, selectMetric, show } = props;
-  const [searchText, setSearchText] = useState("")
-  if(!show){
-    return null;
+function sortDictionary(dictionary){
+    var sortedTagsArr = Object.entries(dictionary).sort((a,b) => b[1].length - a[1].length)
+    var retArr = [];
+    for(let i = 0; i < sortedTagsArr.length; i++){
+      var tempArr = sortedTagsArr[i]
+      var tempDict = {};
+      tempDict[tempArr[0]] = tempArr[1]
+      retArr.push(tempDict)
+    }
+    return retArr
+  }
+  
+//Create a dictionary of all unique tags/datasets that exist, each key holding an array of metrics belonging to that tag/dataset
+function createTagDictionary(supportedMetrics){
```

Comment:
Later we can revise this method to be more runtime efficient. I left a comment below about correctness.